icefall/diagnostics.py

```diff
@@ -28,51 +29,67 @@ class TensorDiagnosticOptions(object):
 
     Args:
       memory_limit:
-        The maximum number of bytes per tensor (limits how many copies
-        of the tensor we cache).
+        The maximum number of bytes per tensor 
+        (limits how many copies of the tensor we cache).
+      max_eig_dim:
+        The maximum dimension for which we print out eigenvalues
+        (limited for speed reasons).
     """
 
-    def __init__(self, memory_limit: int):
+    def __init__(
+        self, 
+        memory_limit: int = (2 ** 20),
+        max_eig_dim: int = 512
+    ):
         self.memory_limit = memory_limit
+        self.max_eig_dim = max_eig_dim
 
     def dim_is_summarized(self, size: int):
         return size > 10 and size != 31
 
 
-def get_sum_abs_stats(
+def get_tensor_stats(
     x: Tensor, dim: int, stats_type: str
 ) -> Tuple[Tensor, int]:
-    """Returns the sum-of-absolute-value of this Tensor, for each index into
-    the specified axis/dim of the tensor.
+    """
+    Returns the specified transformation of the Tensor (either x or x.abs()
+    or (x > 0), summed over all but the index `dim`.
 
     Args:
-      x:
-        Tensor, tensor to be analyzed
-      dim:
-        Dimension with 0 <= dim < x.ndim
+       x: Tensor, tensor to be analyzed
```

Comment:
Please use the previous style for docstring.

See https://google.github.io/styleguide/pyguide.html#383-functions-and-methods for more details.